framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1242,15 +1242,19 @@ public AnnotationMirror createIntRangeAnnotation(long from, long to) {
     }
 
     /**
-     * Create an {@code @IntRange} annotation from the range. May return BOTTOMVAL or UNKNOWNVAL.
+     * Create an {@code @IntRange} or {@code @IntVal} annotation from the range. May return
+     * BOTTOMVAL or UNKNOWNVAL.
      */
     public AnnotationMirror createIntRangeAnnotation(Range range) {
         if (range.isNothing()) {
             return BOTTOMVAL;
         } else if (range.isEverything()) {
             return UNKNOWNVAL;
-        } else {
+        } else if (range.isWiderThan(MAX_VALUES)) {
             return createIntRangeAnnotation(range.from, range.to);
+        } else {
```

Comment:
1.  Annotations are written more compactly and easier to read.  `@IntRange(from=1, to=2)` converted to `@IntVal({1, 2})`

2. If the Range was from an explicitly written `@IntVal`, then it will appear in error messages as the user wrote it.

3.  At binary comparisons, such as `<`, if one side is an `@IntRange` and the other is `@IntVal`, the `@IntVal` is converted to a range.  This causes the refinement of `@IntVal` side to be less precise.  For example,

```
void test(@IntVal({1, 10}) int value, @IntRange(from=3, to=7) int range) {
  if (value < range) {
    // without this conversion: value: @IntVal({1, 10})
    // with this conversion: value: @IntVal(1)
    // in both cases range isn't refined and therefor is still @IntRange(from=3, to=7)
  }
```
So this conversion makes the Value Checker a little more precise without having to write refinement rules when one annotation is a range and the other is a value.  The Transfer function could do this conversion just before refining types and get the same benefit.